### PR TITLE
Feature/last sequence number refactor

### PR DESCRIPTION
**Pull request by humanjas**
#88 

# Introduction
The current implementation of the new sequence number of a block is very dirty and needs to be implemented differently.

# Purpose
It is implemented well now

# Future Applicability
The BlockFactory class will need to be used by the blockcontroller and not by other classes.

# PR Reflection

### CODE REVIEW

File: app/src/main/java/nl/tudelft/b_b_w/controller/Api.java [hunk 1]
```diff
@@ -79,7 +79,17 @@ public final void addKey(String owner, String user, String key, String iban) thr
                 key, prevHashSelf, prevHashOther, iban);
         String hash = conversionController.hashKey();
 
-        Block fresh = BlockFactory.getBlock("BLOCK", owner, hash, prevHashSelf, prevHashOther, key, iban, 0);
+        Block fresh = BlockFactory.getBlock(
+                "BLOCK",
```
Comment (ymte) [inline app/src/main/java/nl/tudelft/b_b_w/controller/Api.java]: Hardcoded values are a big no-no.

File: app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java [hunk 2]
```diff
@@ -7,62 +7,78 @@
...
  * Performs the actions on the blockchain
  */
 
-public class BlockController {
-    /** Context of the block database */
+public class BlockController implements BlockControllerInterface {
+    /**
+     * For when info is not available.
+     */
+    private static final String NA = "N/A";
```
Comment (ymte) [inline app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java]: Good use of constants.

File: app/src/main/java/nl/tudelft/b_b_w/controller/Api.java [hunk 3]
```diff
@@ -109,7 +119,17 @@ public final void revokeKey(String owner, String user, String key, String iban)
                 key, prevHashSelf, prevHashOther, iban);
         String hash = conversionController.hashKey();
 
-        Block fresh = BlockFactory.getBlock("REVOKE", owner, hash, prevHashSelf, prevHashOther, key, iban, 0);
+        Block fresh = BlockFactory.getBlock(
+                "REVOKE",
```
Comment (ymte) [inline app/src/main/java/nl/tudelft/b_b_w/controller/Api.java]: Same here.

File: app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java [hunk 4]
```diff
@@ -74,34 +90,32 @@ public final boolean blockExists(String owner, String key, boolean revoked) {
...
      */
+    @Override
     public final void clearAllBlocks() {
-        databaseHandler.clearAllBlocks();
+        mutateDatabaseHandler.clearAllBlocks();
     }
 
     /**
-     * Get all blocks that are not revoked in sorted order
-     *
```
Comment (deepweeb) [inline app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java]: You missed the @param here

Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java]: That is the old version

**Comment by humanjas** [comment]
Changed it

**Pull request merged**